### PR TITLE
Remove SimpleSmt store

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

This is just a small simplification to the codebase. ATM the Store is not useful for much, and adds an extra layer of indirection that seems unnecessary for the time being.

We can add this later on if necessary (or not merge this, it is just a suggestion and it was a quick one to make)

edit: the original title was super confusing, I didn't mean the structure named `MerkleStore`, I meant the struct `Store` inside the `merkle::siple_smt` module. Sorry for any confusion, I have renamed the PR.

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: src/merkle/simple_smt/mod.rs [hunk 1]
```diff
@@ -9,13 +9,20 @@ mod tests;
...
 /// The root of the tree is recomputed on each new leaf update.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct SimpleSmt {
     root: Word,
+    leaves: BTreeMap<u64, Word>,
     depth: u8,
-    pub(crate) store: Store,
+    empty_hashes: Vec<RpoDigest>,
+    pub(crate) branches: BTreeMap<NodeIndex, BranchNode>,
+}
```
Comment (bobbinth) [inline src/merkle/simple_smt/mod.rs:20]: minor nit: I'd probably change the order here to be like:
```Rust
pub struct SimpleSmt {
    depth: u8,
    root: Word,
    leaves: BTreeMap<u64, Word>,
    pub(crate) branches: BTreeMap<NodeIndex, BranchNode>,
    empty_hashes: Vec<RpoDigest>,
}
```
Also, would be nice to make `branches` private - but maybe in a different PR.

File: src/merkle/simple_smt/mod.rs [hunk 2]
```diff
@@ -177,67 +189,26 @@ impl SimpleSmt {
...
-            branches,
-            leaves,
-            empty_hashes,
-            depth,
-        };
-
-        (store, root)
-    }
 
     fn replace_empty_subtrees(&mut self, hashes: Vec<RpoDigest>) {
```
Comment (bobbinth) [inline src/merkle/simple_smt/mod.rs:213]: Let's add a section heading above this line. Could be:
```
// HELPER METHODS
// --------------------------------------------------------------------------------------------
```

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left a couple of minor comments inline. Once these are done we can merge.

File: src/merkle/simple_smt/mod.rs (hunk 2 above)
Comment (hackaugusto) [reply src/merkle/simple_smt/mod.rs:216]: added

File: src/merkle/simple_smt/mod.rs (hunk 1 above)
Comment (hackaugusto) [reply src/merkle/simple_smt/mod.rs]: changed fields order.

making the `branches` private requires a different way of getting to the nodes, something like https://github.com/0xPolygonMiden/crypto/pull/117 would do. (I started this cleanup as I was reading the code to figure out how much work would that be).

I agree it should be done on separate PR though :)

**Pull request merged**